### PR TITLE
Change from setup.cfg to pyproject.toml

### PR DESCRIPTION
**Pull request by wch**
This PR switches htmltools from setup.cfg/setup.py to the newer pyproject.toml config file.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    "pyright>=1.1.334",
+    "pre-commit>=2.15.0",
+    "wheel",
+    "build"
+]
+
+[tool.setuptools]
+packages = ["htmltools"]
+include-package-data = true
+# test-suite = "tests"
```
[the remark this answers is not in the document]
Comment (machow) [reply pyproject.toml:59]: I would set any test options using [a pytest section](https://docs.pytest.org/en/stable/reference/customize.html#pyproject-toml)

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+[build-system]
+requires = [
+  "setuptools",
+  "wheel"
+]
+build-backend = "setuptools.build_meta"
```
Comment (machow) [inline pyproject.toml:6]: If you use [setuptools-scm](https://github.com/pypa/setuptools_scm/), you can set the version automatically using git tags

(but the current approach seems totally fine and reasonable!)

**Review by machow — APPROVED** [review]
LGTM -- I didn't test out building the package, only read through (and have to admit that I can never remember the names of things in pyproject.toml files)

**Comment by wch** [comment]
One unfortunate thing about this change is that it apparently breaks pyright support when this py-htmltools is installed as an editable package with `pip install -e`.

https://github.com/microsoft/pyright/issues/3846
https://github.com/microsoft/pyright/issues/6209

It looks like the workaround is to install with `editable_mode`:

```
pip install -e . --config-settings editable_mode=strict
# or
pip install -e . --config-settings editable_mode=compat
```

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (wch) [reply pyproject.toml:59]: Thanks, `pytest` seems to work without this line, so I'll just remove it.

File: pyproject.toml (hunk 2 above)
Comment (wch) [reply pyproject.toml:6]: Thanks - maybe in the future we'll switch over.

**Pull request merged**